5.0 Draft/IDD/IDDs Service Registry/eu.arrowhead.service-discovery-http-json.yml

```diff
@@ -0,0 +1,631 @@
+        the record limit is imposed.
+      parameters:
+        - $ref: '#/components/parameters/query-limit'
+        - $ref: '#/components/parameters/query-offset'
+        - $ref: '#/components/parameters/query-q'
+      responses:
+        '200':
+          description: Matching service records.
+          content:
+            application/json:
+              schema:
+                type: array
+                items: { $ref: '#/components/schemas/Record' }
+          headers:
+            Cache-Control:
+              $ref: '#/components/headers/Cache-Control'
+            Limit:
+              $ref: '#/components/headers/Limit'
+        '400': { $ref: '#/components/responses/400-BadRequest-ExpressionIsInvalid' }
+        '401': { $ref: '#/components/responses/401-Unauthorized' }
+        '403': { $ref: '#/components/responses/403-Forbidden' }
+        '408': { $ref: '#/components/responses/408-RequestTimeout' }
+        '414': { $ref: '#/components/responses/414-URITooLarge' }
+        '415': { $ref: '#/components/responses/415-UnsupportedMediaType' }
+        '429': { $ref: '#/components/responses/429-TooManyRequests' }
+        '431': { $ref: '#/components/responses/431-RequestHeaderFieldsTooLarge' }
+        '500': { $ref: '#/components/responses/500-InternalServerError' }
+        '503': { $ref: '#/components/responses/503-ServiceUnavailable' }
+
+    patch:
+      operationId: register
+      description: |
+        Registers or updates one or more records.
+
+        The combination of the provider and type identifiers inside each record must be unique.
+        Providing a record when a corresponding combination of identifiers are already registered
+        will be interpreted as an attempt to replace the existing record. If more than one record
+        with corresponding identifiers are in the request, the last one in its payload will replace
+        all others.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: array
+              items: { $ref: '#/components/schemas/Record' }
+      responses:
+        '204':
+          description: Confirms that the records in question have been registered or updated.
+        '400': { $ref: '#/components/responses/400-BadRequest-RecordIsInvalid' }
+        '401': { $ref: '#/components/responses/401-Unauthorized' }
+        '403': { $ref: '#/components/responses/403-Forbidden' }
+        '406': { $ref: '#/components/responses/406-NotAcceptable' }
+        '408': { $ref: '#/components/responses/408-RequestTimeout' }
+        '411': { $ref: '#/components/responses/411-LengthRequired' }
+        '413': { $ref: '#/components/responses/413-PayloadTooLarge' }
+        '431': { $ref: '#/components/responses/431-RequestHeaderFieldsTooLarge' }
+        '500': { $ref: '#/components/responses/500-InternalServerError' }
+        '503': { $ref: '#/components/responses/503-ServiceUnavailable' }
+
+    delete:
+      operationId: deregister
```

Comment:
DELETE should be on the single resource i.e. `/services/{service-id}`, in REST this now means "delete the entire service collection" (i.e. all service registrations).

Reply:
It does, which is why you have to use the `deregister-q` query parameter to limit what records are deleted.